pkg/proxy/proxyinfo.go

```diff
@@ -30,8 +30,31 @@ type sidecarSyncStatus struct {
 	xds.SyncStatus
 }
 
+type NodeType string
+
+// Info contains the version for a single data plane component
+type Info struct {
+	ID           string
+	IstioVersion string
+	Type         NodeType
+}
+
+func ToUserFacingNodeType(t model.NodeType) NodeType {
+	switch t {
+	case model.Router:
+		return "gateway"
+	case model.Ztunnel, model.Waypoint, model.SidecarProxy:
+		return NodeType(t)
+	}
+	return ""
+}
+
+func (n NodeType) EqualsModelNodeType(t model.NodeType) bool {
+	return n == ToUserFacingNodeType(t)
+}
```

Comment:
what is this for?

Reply:
Removed. It's unused code. Also, reverted back `ProxyInfo` to `version.go` since there's a strange import cycle...